### PR TITLE
Add service_account.Credentials.to_jwt_credentials

### LINKED ISSUE #35: Implement service_account.Credentials.to_jwt_credentials()
(Context: #29)

### PR DESCRIPTION
**Pull request by jonparrott**
Closes #35.

### CODE REVIEW

File: google/oauth2/service_account.py [hunk 1]
```diff
@@ -204,6 +204,33 @@ def from_service_account_file(cls, filename, **kwargs):
             filename, require=['client_email', 'token_uri'])
         return cls._from_signer_and_info(signer, info, **kwargs)
 
+    def to_jwt_credentials(self):
+        """Creates a :cls:`google.auth.jwt.Credentials` instance from this
```
Comment (dhermes) [inline google/oauth2/service_account.py]: cls a synonym for class? Why don't I **ALWAYS** use `cls`!

File: google/oauth2/service_account.py [hunk 2]
```diff
@@ -204,6 +204,33 @@ def from_service_account_file(cls, filename, **kwargs):
...
+            jwt_from_svc = svc_credentials.to_jwt_credentials()
+            # is the same as:
+            jwt_creds = jwt.Credentials.from_service_account_file(
+                'service_account.json')
+
+        Returns:
+            google.auth.jwt.Credentials: A new Credentials instance.
+        """
+        return jwt.Credentials(
+            self._signer,
```
Comment (dhermes) [inline google/oauth2/service_account.py]: Is there a worry here of one creds instance mutating the signer on purpose, which well make the other creds instance unhappy? (i.e. shouldn't you copy the signer?)

File: google/oauth2/service_account.py [hunk 3]
```diff
@@ -204,6 +204,33 @@ def from_service_account_file(cls, filename, **kwargs):
             filename, require=['client_email', 'token_uri'])
         return cls._from_signer_and_info(signer, info, **kwargs)
 
+    def to_jwt_credentials(self):
```
Comment (dhermes) [inline google/oauth2/service_account.py]: Add an optional `audience`? Or the class is already good at handling that?

File: tests/oauth2/test_service_account.py [hunk 4]
```diff
@@ -58,15 +58,14 @@ def credentials_fixture(self, signer):
             signer, self.SERVICE_ACCOUNT_EMAIL, self.TOKEN_URI)
 
     def test_from_service_account_info(self):
-        with open(SERVICE_ACCOUNT_JSON_FILE, 'r') as fh:
-            info = json.load(fh)
```
Comment (dhermes) [inline tests/oauth2/test_service_account.py]: Pete and re-pete were on a boat?

You seem to like `info = SERVICE_ACCOUNT_INFO.copy()` everywhere else (would've also been a smaller diff, "diff-driven development")

File: google/oauth2/service_account.py (hunk 2 above)
Comment (jonparrott) [inline google/oauth2/service_account.py]: Signer is immutable.

File: tests/oauth2/test_service_account.py (hunk 4 above)
Comment (jonparrott) [inline tests/oauth2/test_service_account.py]: copy() is used when I modify the info to induce particular behavior.

File: google/oauth2/service_account.py (hunk 3 above)
Comment (jonparrott) [inline google/oauth2/service_account.py]: `with_claims` covers it.

**Pull request merged**